realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -45,7 +45,7 @@
  * @see io.realm.Realm
  * @see io.realm.DynamicRealm
  */
-abstract class BaseRealm implements Closeable {
+public abstract class BaseRealm implements Closeable {
```

Comment:
After talking to @beeender We decided we can pull BaseRealm from the public API after all. Only downside is adding a little bit of complexity to BaseRealmAdapter.